### DISCUSSION TITLE
Provide tasmota-tls build on release

### DISCUSSION THREAD

**Discussion by jeroenst** [arendst/Tasmota#13489] [Ideas]
I'm compiling tasmota tls my self at this moment, which works ok. 

But would it be possible to add a tasmota-tls binary to the future releases?

**Comment by sfromis** [comment]
TLS is not a type of feature well-suited for separate binaries, as every binary could have a TLS and a non-TLS version, and TLS is not a common feature to use.

However, the tasmota-zbbridge binary should have TLS included.

**Comment by jeroenst** [comment]
True, but it doesn't have to be compiled for each binary imo. Only the tasmota binary would be a lot of help.

**Comment by s-hadinger** [comment]
Even worse, it's currently two different binaries whether you want fingerprint validation or CA validation. I will probably make it such you can you only one for both. We used to be super short on flash and memory, it's slightly more relaxed now

**Comment by jeroenst** [comment]
I understand. Although in my case CA validation or fingerprint is not mandatory for me, I think the man in the middle attack is very low risk for my appliances. Fingerprint validation fails at my site when the certificate changes so for me not useable. I have a valid CA but there are of course people who use tls without ca.

**Reply by sfromis** [reply]
If you use the tasmota-zbbridge binary (you need no ZbBridge for that), you already have your TLS precompiled build, but thats only with **one** type of setup. It is just not well-suited for precompiled binaries when some types of configuration needs to happen at build time.

**Comment by s-hadinger** [comment]
Zbbridge has CA validation. Letsencrypt and AWS IOT CAs are recognized.

**Comment by jeroenst** [comment]
Maybe it is possible to disable ca validation in the console?
